File: backend/notification-service/src/constants/inAppNotificationTypes.js
/**
 * Types affichés dans le centre de notifications utilisateur (mobile + cloche).
 * Exclut crash, erreurs techniques et notifications système — réservées au backoffice.
 */
const IN_APP_NOTIFICATION_TYPES = [
  'REMINDER',
  'APPLICATION_UPDATE',
  'INTERVIEW_SCHEDULED',
  'FOLLOWUP_DUE',
  'DEADLINE',
  'STATUS_CHANGE',
];

const NON_IN_APP_NOTIFICATION_TYPES = [
  'SYSTEM',
  'CRASH_REPORT',
  'ERROR_REPORT',
];

function resolveNotificationScope(query = {}) {
  const scope = String(query.scope || 'in_app').toLowerCase();
  return scope === 'all' ? 'all' : 'in_app';
}

function buildInAppTypeFilter(scope) {
  if (scope === 'all') return {};
  return { type: { in: IN_APP_NOTIFICATION_TYPES } };
}

module.exports = {
  IN_APP_NOTIFICATION_TYPES,
  NON_IN_APP_NOTIFICATION_TYPES,
  resolveNotificationScope,
  buildInAppTypeFilter,
};

File: mobile/lib/utils/in_app_notification_types.dart
/// Types métier affichés dans le centre de notifications (candidatures, relances, entretiens…).
/// Exclut crash, erreurs techniques et notifications système (backoffice uniquement).
const inAppNotificationTypes = <String>{
  'REMINDER',
  'APPLICATION_UPDATE',
  'INTERVIEW_SCHEDULED',
  'FOLLOWUP_DUE',
  'DEADLINE',
  'STATUS_CHANGE',
};

bool isInAppNotificationType(String type) =>
    inAppNotificationTypes.contains(type.toUpperCase());

List<T> filterInAppNotifications<T>(
  Iterable<T> items,
  String Function(T item) typeOf,
) =>
    items.where((item) => isInAppNotificationType(typeOf(item))).toList();
